crypt.go

```diff
@@ -478,6 +394,60 @@ type ActivateWithTPMSealedKeyOptions struct {
 	KeyringPrefix string
 }
 
+type KeyUnsealer interface {
+	UnsealKey(volumeName, sourceDevicePath string, p Prompter) (key, resealToken []byte, err error)
+	UnderstoodError(e error, isCryptsetupError bool) (ok bool, reason RecoveryKeyUsageReason, err error)
```

Comment:
I would like to avoid UnderstoodError myself OTOH it's  a bit unclear how much work it is, and whether it's premature to design those implementation agnostic errors. Most of the errors the code seems to care about are TPM specific and what we need more than errors is what to pass to the recovery key using code.